cmd/kubeadm/app/constants/constants.go

```diff
@@ -366,6 +366,19 @@ const (
 	// May be overridden by a flag at startup.
 	// Deprecated: use the secure KubeControllerManagerPort instead.
 	InsecureKubeControllerManagerPort = 10252
+
```

Comment:
This stuff should really be in the component config for the apiserver.  
/cc @mtaufen @luxas @stealthybox

Reply:
agree. was there a reason kube-apiserver didn't get an exported config package along with scheduler/kubelet/controller-manager?

Reply:
that said, I also would not be bothered by use of literal strings in `getAuthzModes`. those modes cannot change without breaking consumers.

Reply:
given the diff already has a TODO for this above the constants, i think we should proceed with this temporary solution.

/lgtm
@rosti please remove the hold if you think this is OK.

Reply:
I agree. Let's get it merged as is.
Thank you for the clarification @liggitt !
/hold cancel